builder/builder.go

```diff
@@ -62,91 +87,60 @@ func FromBuild(build *api.Build) Builder {
 	}
 }
 
-// NewBuild is the API entry for the BuilderScheduler. Create a new Build instance based on PlatformBuild.
-func NewBuild(platformBuild api.PlatformBuild, publishImage string, buildName string) Scheduler {
-	// TODO: Figure if we need a PlatformBuild builder fluent api.
-	// TODO: Verify structure integrity
+// NewBuild is the API entry for the Builder. Create a new Build instance based on PlatformBuild.
+func NewBuild(info BuilderInfo) Scheduler {
 	ctx := buildContext{
-		BaseImage: platformBuild.Spec.BaseImage,
+		BaseImage: info.Platform.Spec.BaseImage,
 		C:         context.TODO(),
 	}
 
-	if platformBuild.Spec.BuildStrategy == api.BuildStrategyPod && platformBuild.Spec.PublishStrategy == api.PlatformBuildPublishStrategyKaniko {
-		ctx.Build = api.NewKanikoBuild(platformBuild, publishImage, buildName)
-	} else {
-		panic(fmt.Errorf("BuildStrategy %s with PublishStrategy %s is not supported", platformBuild.Spec.BuildStrategy, platformBuild.Spec.PublishStrategy))
+	for _, v := range schedulers {
+		if v.CanHandle(info) {
+			return v.CreateScheduler(info, ctx)
+		}
 	}
+	panic(fmt.Errorf("BuildStrategy %s with PublishStrategy %s is not supported", info.Platform.Spec.BuildStrategy, info.Platform.Spec.PublishStrategy))
+}
 
-	return &scheduler{
-		builder: builder{
-			L:       log.WithName(util.ComponentName),
-			Context: ctx,
-		},
-		Resources: make([]resource, 0),
-	}
+func (s *scheduler) WithClient(client client.Client) Scheduler {
+	s.builder.WithClient(client)
+	return s.Scheduler
 }
 
-func (b *scheduler) WithClient(client client.Client) Scheduler {
-	b.builder.WithClient(client)
-	return b
+func (s *scheduler) WithResource(target string, content []byte) Scheduler {
+	s.Resources = append(s.Resources, resource{target, content})
+	return s.Scheduler
 }
 
-func (b *scheduler) WithResource(target string, content []byte) Scheduler {
-	b.Resources = append(b.Resources, resource{target, content})
-	return b
+func (s *scheduler) WithResourceRequirements(res corev1.ResourceRequirements) Scheduler {
+	// no default implementation.
+	return s.Scheduler
+}
+
+func (s *scheduler) WithAdditionalArgs(args []string) Scheduler {
+	// no default implementation.
+	return s.Scheduler
+}
+
+func (s *scheduler) WithProperty(property BuilderProperty, object interface{}) Scheduler {
+	// no default implementation
+	return s.Scheduler
 }
 
 // Schedule schedules a new build in the platform
-func (b *scheduler) Schedule() (*api.Build, error) {
+func (s *scheduler) Schedule() (*api.Build, error) {
```

Comment:
how this scheduler works?
seems it only mounts a configMap a reconcile.

maybe better change the method name?

Reply:
yeah, for now it's just that. It schedules the build by mounting the volume. This is not in scope for this PR.